config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -101,3 +106,232 @@ const (
+// - "NotDeployed": The operator will not deploy an Alertmanager instance.
+type AlertManagerDeployMode string
+
+const (
+	// AlertManagerModeEnabled means the Alertmanager instance will be deployed and managed by the operator.
+	AlertManagerDeployModeDeployed AlertManagerDeployMode = "Deployed"
+
+	// AlertManagerModeDisabled means the operator will not deploy the Alertmanager instance.
+	AlertManagerDeployModeNotDeployed AlertManagerDeployMode = "NotDeployed"
+)
+
+// UserAlertManagerMode defines mode for user-defines namespaced
+//
+// Possible values:
+// - "Selectable": User-defined namespaces can be selected for AlertmanagerConfig lookups.
+// - "None": User-defined namespaces cannot be selected for AlertmanagerConfig lookups.
+type UserAlertManagerMode string
+
+const (
+	// UserAlertmanagerEnabled enables user-defined namespaces to be selected for `AlertmanagerConfig` lookups. This setting only
+	// applies if the user workload monitoring instance of Alertmanager is not enabled.
+	UserAlertManagerSelectable UserAlertManagerMode = "Selectable"
+	// UserAlertManagerDisabled disables user-defined namespaces to be selected for `AlertmanagerConfig` lookups. This setting only
+	// applies if the user workload monitoring instance of Alertmanager is not enabled.
+	UserAlertManagerNone UserAlertManagerMode = "None"
+)
+
+// logLevel defines the verbosity of logs emitted by Alertmanager.
+// Valid values are Error, Warn, Info and Debug.
+// +kubebuilder:validation:Enum=Error;Warn;Info;Debug
+type LogLevel string
+
+const (
+	LogLevelError LogLevel = "Error"
+	LogLevelWarn  LogLevel = "Warn"
+	LogLevelInfo  LogLevel = "Info"
+	LogLevelDebug LogLevel = "Debug"
+)
+
+// ResourceSpec defines the requested and limited value of a resource.
+type ResourceSpec struct {
+	// request is the minimum amount of the resource required (e.g. "2Mi", "1Gi").
+	// +kubebuilder:validation:MaxLength=24
+	// This filed is optional
+	// +optional
+	Request string `json:"request,omitempty"`
+	// limit is the maximum amount of the resource allowed (e.g. "2Mi", "1Gi").
+	// +kubebuilder:validation:MaxLength=24
+	// This filed is optional
+	// +optional
+	Limit string `json:"limit,omitempty"`
+}
+
+// AlertmanagerContainerResources defines simplified resource requirements for a container.
+type AlertmanagerContainerResources struct {
+	// cpu defines the CPU resource limits and requests.
+	// This filed is optional
+	// +optional
+	CPU *ResourceSpec `json:"cpu,omitempty"`
+
+	// memory defines the memory resource limits and requests.
+	// This filed is optional
+	// +optional
+	Memory *ResourceSpec `json:"memory,omitempty"`
+
+	// hugepages is a list of hugepage resource specifications by page size.
+	// defines an optional list of unique configurations identified by their `size` field.
+	// A maximum of 10 items is allowed.
+	// The list is treated as a map, using `size` as the key, which simplifies updates and replacements
+	// of individual entries.
```

Comment:
For an end-user, I believe this means that entries must be unique. I'm not sure an end-user cares about whether or not this simplifies updates and replacements of individual entries.